docs/storage/lumio/advanced.md

```diff
@@ -0,0 +1,514 @@
+locations under home. This can be achieved using environment variables or command line flags.
+
+
+|      | rclone        | s3cmd                  | aws                                         |
+|------|---------------|------------------------|---------------------------------------------|
+| DEFAULT | `~/.config/rclone/rclone.conf`   | `~/.s3cfg`  | `~/.aws/credentials` and `~/.aws/config`|
+| ENV  | `RCLONE_CONFIG` | `S3CMD_CONFIG`           | `AWS_SHARED_CREDENTIALS_FILE` and `AWS_CONFIG_FILE` |
+| FLAG | `--config FILE` | `-c FILE`, `--config=FILE` |                                             |
+
+The `aws` cli additionally has the concept of [profiles](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html), and you can specify 
+which one to use using the `--profile <name>` flag or the `AWS_PROFILE` environment variable.
+
+### Environment
+
+Most programs will use the environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`
+when trying to authenticate. So these can be set if one does not wish to save the credentials on disk.
+The environment variables do not always take precedence over values set in configuration files, as
+is the case for `s3cmd` and `rclone`. This means that invalid credentials in config files will
+lead to an access denied even if there are valid credentials in the environment. The `aws` command
+will use the environment variables instead of `~/.aws/credentials` if they are set. 
+`rclone` will additionally require `RCLONE_S3_ENV_AUTH=true` in the environment or `env_auth = true`
+in the config file.
+
+
+## Programmatic access 
+
+When use cases become sufficiently complex one might want to interact
+with LUMI-O in a more programmatic fashion instead of using the 
+command line tools. One such option is the AWS SDK for Python [boto3](https://pypi.org/project/boto3/)\*.
+
+The script
+
+```python
+import boto3
+
+session = boto3.session.Session(profile_name='lumi-465000001')
+s3_client = session.client('s3')
+buckets=s3_client.list_buckets()
+```
+
+Would fetch the buckets of project 465000001 and return the information as a python dictionary. 
+For the full list of available functions, see the [aws s3 client documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html)
+
+
+If a default profile has been configured `~/.aws/credentials`
+the client creation can be shortened to:
+
+```python
+import boto3
+s3_client = boto3.client('s3')
+```
+
+boto3 uses the same configuration files and respects the same environment variables as the `aws` cli.
+
+!!! note 
+	You will need a sufficiently new version of boto3 (e.g 1.26 which is installed if using python3.6 is to old) for it
+	to understand a default profile set in ~/.aws/credentials and corresponding config file, 
+	otherwise the tool will always default to aws s3 endpoint and you will need to specify the
+	profile/endpoint when constructing the client.
```

Comment:
```suggestion
	You will need a sufficiently new version of boto3 (e.g version 1.26, which is installed if using python3.6, is too old) for it
	to understand a default profile set in ~/.aws/credentials and corresponding config file, 
	otherwise the tool will always default to aws s3 endpoint and you will need to specify the
	profile/endpoint when constructing the client.
```

Reply:
Now this is correct interpretation, right? Version 1.26 of boto3 would be too old. (And it's the version that you get if you use python 3.6.)

Reply:
yes